gnovm/pkg/gnolang/values.go

```diff
@@ -1521,6 +1536,37 @@ func (tv *TypedValue) ComputeMapKey(store Store, omitType bool) MapKey {
 	return MapKey(bz)
 }
 
+// - PrimitiveValue is moveable across realm boundary.
+// - Array and Struct are moveable across realm only if they are composed of moveable types(recursively applied). The value will be copied to the callee realm.
+// - Slice are moveable across realm only if they are referring to moveable types(recursively applied)(not implemented yet).
+// - Any other cases require some complex inter-realm object management scheme, low priority.
+func (tv *TypedValue) IsMoveableAcrossRealm() bool {
```

Comment:
This is a misnomer because there is already a notion of what is passable between realms as already implemented... that is, one can already call a realm function from another realm, passing in objects, as long as objects aren't sharing persistence. What is implemented here conflicts with that.

I think what you maybe want is IsMoveableAcrossChains or IsIBCArgument, and for that for now I think it should be limited to a list of stringified primitives only, no structs;

and then we need to have a whole series of conversations to evolve that into something that either supports the EVM ABI, or some other interop standard.